pkg/test/spaceprovisionerconfig/spaceprovisionerconfig_assertions.go

```diff
@@ -0,0 +1,44 @@
+package spaceprovisionerconfig
+
+import (
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/condition"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test/assertions"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type predicate func(*toolchainv1alpha1.SpaceProvisionerConfig) bool
+
+var _ assertions.Predicate[*toolchainv1alpha1.SpaceProvisionerConfig] = predicate(nil)
+
+func (p predicate) Matches(obj *toolchainv1alpha1.SpaceProvisionerConfig) bool {
+	return p(obj)
+}
```

Comment:
btw, there is one disadvantage of using the same type predicate that returns just a boolean at both places - it's hard to understand why the unit test failed - there is no assertion message, that something was expected but the actual value is different

Reply:
Anyway, let's not try to solve all the problems of the universe in one PR - let's merge it as it is and let's see how this will evolve and is gonna be used

Reply:
That is true... There are 2 ways of addressing it:
1. Change `AssertThat` to accept `msgAndArgs ...any` and pass it to the inner `assert.True` so that the caller has some control over the output of the asserts.
2. Change the `Predicate.Matches` to return `(bool, string)` where the second parameter could contain a description of why the predicate didn't match. This could then somehow be incorporated into `AssertThat` and `wait.For().FirstThat()`.

I will update this PR with the former, but I would wait with the latter until we find we actually need it. WDYT?

Reply:
Sounds good :+1: